File: BaekjoonOnlineJudge/DFS/연결요소의 개수 (11724)/Solution.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> graph[1001];
bool visited[1001];

bool dfs(int x){
	if (visited[x])
		return false;

	visited[x] = true;

	for (int i = 0;i < graph[x].size();i++) {
		int y = graph[x][i];
		if (!visited[y]) dfs(y);
	}

	return true;
}

int main() {
	int n, m, result = 0;
	cin >> n >> m;

	for (int i = 0;i < m;i++) {
		int a, b;
		cin >> a >> b;
		graph[a].push_back(b);
		graph[b].push_back(a);
	}

	for (int i = 1;i <= n;i++) {
		if (dfs(i))
			result++;
	}

	cout << result;
}

File: BaekjoonOnlineJudge/DFS/연결요소의 개수 (11724)/Solution.py
import sys
sys.setrecursionlimit(10000)

n, m = map(int, input().split())
graph = [[] for _ in range(n+1)]
visited = [False] * (n+1)
for _ in range(m):
  a, b = map(int, sys.stdin.readline().split())
  graph[a].append(b)
  graph[b].append(a)


def dfs(start):
  visited[start] = True
  for node in graph[start]:
    if not visited[node]:
      dfs(node)


result = 0
for i in range(1, n+1):
  if not visited[i]:
    dfs(i)
    result += 1
print(result)

'''
만나서 더러웠고 다신 보지말자^-^!
'''
